### ISSUE TITLE
@timestamp not work

### ISSUE THREAD

**Issue by LittlePeng** [uken/fluent-plugin-elasticsearch#157]
after i add a `@timestamp` filed with milisecond to the record like:
{"@timestamp":"2016-03-12 20:25:24.770", ....}

and add `time_key_format %Y-%m-%d %H:%M:%S.%N`, then logs can not send to elasticsearch and noting error msg.

```
      @type elasticsearch
      host 192.168.200.20
      port 9200
      logstash_format true # defaults to false
      time_key_format %Y-%m-%d %H:%M:%S.%N
      logstash_prefix error
      index_name fluentd
      type_name fluentd
```

**Comment by pitr** [comment]
Could you try replicating the issue by modifying [this setup](https://gist.github.com/pitr/9a518e840db58f435911) and publishing as gist? For step 3 you can just do `bin/fluentd --config fluent.conf`

**Comment by LittlePeng** [comment]
```
/opt/td-agent/embedded/bin/fluentd -c fluent.conf
2016-03-16 19:47:41 +0800 [info]: reading config file path="fluent.conf"
2016-03-16 19:47:41 +0800 [info]: starting fluentd-0.12.12
2016-03-16 19:47:42 +0800 [info]: gem 'fluent-mixin-config-placeholders' version '0.3.0'
2016-03-16 19:47:42 +0800 [info]: gem 'fluent-mixin-plaintextformatter' version '0.2.6'
2016-03-16 19:47:42 +0800 [info]: gem 'fluent-plugin-elasticsearch' version '1.4.0'
2016-03-16 19:47:42 +0800 [info]: gem 'fluent-plugin-forest' version '0.3.0'
2016-03-16 19:47:42 +0800 [info]: gem 'fluent-plugin-grep' version '0.3.4'
2016-03-16 19:47:42 +0800 [info]: gem 'fluent-plugin-mongo' version '0.7.10'
2016-03-16 19:47:42 +0800 [info]: gem 'fluent-plugin-rewrite-tag-filter' version '1.4.1'
2016-03-16 19:47:42 +0800 [info]: gem 'fluent-plugin-s3' version '0.5.9'
2016-03-16 19:47:42 +0800 [info]: gem 'fluent-plugin-scribe' version '0.10.14'
2016-03-16 19:47:42 +0800 [info]: gem 'fluent-plugin-td' version '0.10.27'
2016-03-16 19:47:42 +0800 [info]: gem 'fluent-plugin-td-monitoring' version '0.2.1'
2016-03-16 19:47:42 +0800 [info]: gem 'fluent-plugin-webhdfs' version '0.4.1'
2016-03-16 19:47:42 +0800 [info]: gem 'fluentd' version '0.12.12'
2016-03-16 19:47:42 +0800 [info]: adding match pattern="to.es" type="elasticsearch"
2016-03-16 19:47:42 +0800 [info]: adding source type="dummy"
2016-03-16 19:47:42 +0800 [info]: using configuration file: <ROOT>
  <source>
    @type dummy
    tag to.es
    dummy {"hello":"world1111111111","@timestamp":"2016-03-12 20:25:24.770"}
    rate 1000
  </source>
  <match to.es>
    @type elasticsearch
    logstash_format true
    buffer_chunk_limit 64k
    flush_interval 2s
    host 192.168.200.21
    port 9200
    time_key_format %Y-%m-%d %H:%M:%S.%N
    logstash_prefix error
    index_name fluentd
    type_name fluentd
  </match>
</ROOT>
2016-03-16 19:47:43 +0800 [info]: Connection opened to Elasticsearch cluster => {:host=>"192.168.200.21", :port=>9200, :scheme=>"http"}
```
after this kibana  Fatal Error:
```
Courier Fetch: [unsupported_operation_exception] custom format isn't supported
Version: 4.4.1
Build: 9693
```
i will try to find what wrong ...

**Comment by pitr** [comment]
Could it be related to this issue - https://discuss.elastic.co/t/leap-day-issues/43018

**Comment by mpas** [comment]
I seem to have the same issue. No error message and nothing appears in Elastic Search.

Using the following config
```
  <match **>
    @type elasticsearch
    host "#{ENV['ELASTIC_SEARCH_URL']}"
    port "#{ENV['ELASTIC_SEARCH_PORT']}"
    time_key_format %Y-%m-%d %H:%M:%S.%N
    logstash_format true
  </match>
```

The following timestamp is in the record..

`"@timestamp":"2016-06-07 11:56:29.593"`

**Comment by tanner-bruce** [comment]
logstash_format does not work with index_name

**Comment by cosmo0920** [comment]
I've released 1.10.2 and 2.0.0.rc.7 to fix `logstash_format` not to work with index_name. Closing.
